saltproc/simulation.py

```diff
@@ -140,13 +144,69 @@ def store_after_repr(self, after_mats, waste_dict, dep_step):
                     # Save isotope indexes map and units in EArray attributes
                     earr.flavor = 'python'
                     earr.attrs.iso_map = iso_idx
+
+                earr, iso_wt_frac = self._add_missing_nuclides(db, earr, iso_idx, iso_wt_frac)
+
                 earr.append(np.asarray([iso_wt_frac], dtype=np.float64))
                 del iso_wt_frac
                 del iso_idx
         # Also save materials AFTER reprocessing and refill here
         self.store_mat_data(after_mats, dep_step, True)
         db.close()
 
+    def _add_missing_nuclides(self, db, earr, iso_idx, iso_wt_frac):
```

Comment:
This method looks like it does several things. I recommend breaking it up into smaller functions for testing purposes.

Reply:
Okay, I'll see what I can do.